### PR TITLE
F1482 create preview env

### PR DESCRIPTION
**Pull request by ssickles**
Added the update method for preview environments and added the ContextKey to types.Environment.

### CODE REVIEW

File: preview_envs.go [hunk 1]
```diff
@@ -36,3 +47,20 @@ func (pe PreviewEnvs) Create(stackId int64, env *CreatePreviewEnvInput) (*types.
...
+// Update - PUT/PATCH /orgs/:orgName/stacks/:stack_id/preview_envs/:id
+func (pe PreviewEnvs) Update(stackId, envId int64, env *UpdatePreviewEnvInput) (*types.Environment, error) {
+	rawPayload, _ := json.Marshal(env)
+	res, err := pe.Client.Do(http.MethodPut, pe.envPath(stackId, envId), nil, nil, json.RawMessage(rawPayload))
+	if err != nil {
+		return nil, err
+	}
+
+	var updatedEnv types.Environment
+	if err := response.ReadJson(res, &updatedEnv); response.IsNotFoundError(err) {
```
Comment (BSick7) [inline preview_envs.go:60]: You can return ReadJsonPtr here instead of this entire block.

**Review by BSick7 — APPROVED** [review]
Looks good, tiny suggestion for tighter code.

File: preview_envs.go (hunk 1 above)
Comment (ssickles) [reply preview_envs.go]: Done, I keep forgetting about this method. I was a victim of copying from the method above here. I went ahead and updated both methods in this file.

File: types/create_env_run_input.go [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package types
 
 type CreateEnvRunInput struct {
-	IsDestroy bool `json:"isDestroy"`
+	IsDestroy bool     `json:"isDestroy"`
+	AppIds    *[]int64 `json:"appIds"`
```
Comment (BSick7) [inline types/create_env_run_input.go:5]: It's odd to use a pointer to a slice.
Can we drop the pointer?

**Review by BSick7 — APPROVED** [review]
Looks good, one minor suggestion.

File: types/create_env_run_input.go (hunk 2 above)
Comment (ssickles) [reply types/create_env_run_input.go:5]: Would it be obvious to you that an empty list of appIds means "launch them all"? I was trying to indicate that its kind of an optional parameter.

Comment (BSick7) [reply types/create_env_run_input.go]: A slice is nilable. If not specified in json or set to `null`, the slice is `nil`.

Comment (ssickles) [reply types/create_env_run_input.go]: Updated to remove the pointer.

**Pull request merged**